allennlp/modules/token_embedders/token_embedder.py

```diff
@@ -24,3 +27,10 @@ def get_output_dim(self) -> int:
         token.  This is `not` the shape of the returned tensor, but the last element of that shape.
         """
         raise NotImplementedError
+
+    def extend_vocab(self, extended_vocab: Vocabulary, vocab_namespace: Optional[str]):
```

Comment:
Here you say it's optional, but you don't actually give a default of `None`.  Just remove the `Optional`.

Reply:
Actually, I intended to make it optional, it's a typo. The tests are not giving error because the overrided methods are fixing it.